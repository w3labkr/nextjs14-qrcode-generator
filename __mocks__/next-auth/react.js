const useSession = jest.fn(() => ({
  data: null,
  status: "loading",
  update: jest.fn(),
}));

const signIn = jest.fn(() => Promise.resolve({ error: null }));
const signOut = jest.fn(() => Promise.resolve());
const getCsrfToken = jest.fn(() => Promise.resolve("mock-csrf-token"));
const getProviders = jest.fn(() => Promise.resolve({}));
const getSession = jest.fn(() => Promise.resolve(null));

const SessionProvider = ({ children }) => children;

module.exports = {
  useSession,
  signIn,
  signOut,
  getCsrfToken,
  getProviders,
  getSession,
  SessionProvider,
};