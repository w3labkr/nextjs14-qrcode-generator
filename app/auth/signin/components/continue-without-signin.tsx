import { Button } from "@/components/ui/button";

export const ContinueWithoutSignIn = () => {
  return (
    <div className="text-center">
      <Button variant="link" asChild>
        <a href="/">로그인 없이 계속하기</a>
      </Button>
    </div>
  );
};
