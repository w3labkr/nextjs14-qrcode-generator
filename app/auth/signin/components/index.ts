export { GoogleSignInButton } from "./google-signin-button";
export { RememberMeCheckbox } from "./remember-me-checkbox";
export { ContinueWithoutSignIn } from "./continue-without-signin";
export { SignInCard } from "./signin-card";
