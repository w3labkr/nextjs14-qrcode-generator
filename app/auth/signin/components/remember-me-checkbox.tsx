"use client";

import { Checkbox } from "@/components/ui/checkbox";
import { Label } from "@/components/ui/label";

interface RememberMeCheckboxProps {
  checked: boolean;
  onCheckedChange: (checked: boolean) => void;
}

export const RememberMeCheckbox = ({
  checked,
  onCheckedChange,
}: RememberMeCheckboxProps) => {
  return (
    <div className="flex items-center space-x-2">
      <Checkbox
        id="remember-me"
        checked={checked}
        onCheckedChange={(checked) => onCheckedChange(checked === true)}
      />
      <Label
        htmlFor="remember-me"
        className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
      >
        30일 동안 기억하기
      </Label>
    </div>
  );
};
