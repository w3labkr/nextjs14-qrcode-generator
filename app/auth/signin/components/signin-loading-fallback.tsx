import {
  Card,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";

export const SignInLoadingFallback = () => {
  return (
    <Card className="w-full max-w-md">
      <CardHeader className="text-center">
        <CardTitle className="text-2xl">로그인</CardTitle>
        <CardDescription>로딩 중...</CardDescription>
      </CardHeader>
    </Card>
  );
};
