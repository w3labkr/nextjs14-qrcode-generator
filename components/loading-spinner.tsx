export function LoadingSpinner() {
  return (
    <div className="flex min-h-screen items-center justify-center">
      <div 
        className="animate-spin rounded-full h-8 w-8 border-b-2 border-primary"
        role="status"
        aria-label="로딩 중..."
      >
        <span className="sr-only">로딩 중...</span>
      </div>
    </div>
  );
}
