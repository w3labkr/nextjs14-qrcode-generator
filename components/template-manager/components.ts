// Template Manager Components
export { default as TemplateManager } from "./index";
export { default as TemplateList } from "./template-list";
export { default as CreateTemplateDialog } from "./create-template-dialog";
export { default as SaveTemplateButton } from "./save-template-button";
export { default as EditTemplateDialog } from "./edit-template-dialog";
export { default as DeleteTemplateDialog } from "./delete-template-dialog";
export { default as LoadingSkeleton } from "./loading-skeleton";
