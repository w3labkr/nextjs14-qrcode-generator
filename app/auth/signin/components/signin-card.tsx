"use client";

import { useState, useCallback } from "react";
import { useSearchParams } from "next/navigation";
import { GoogleSignInButton } from "./google-signin-button";
import { RememberMeCheckbox } from "./remember-me-checkbox";
import { ContinueWithoutSignIn } from "./continue-without-signin";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { setRememberMeCookie } from "@/lib/auth-utils";

export const SignInCard = () => {
  const searchParams = useSearchParams();
  const callbackUrl = searchParams.get("callbackUrl") || "/dashboard";
  const [rememberMe, setRememberMe] = useState(false);

  const handleGoogleSignIn = useCallback(() => {
    setRememberMeCookie(rememberMe);
  }, [rememberMe]);

  return (
    <Card className="w-full max-w-md">
      <CardHeader className="text-center">
        <CardTitle className="text-2xl">로그인</CardTitle>
        <CardDescription>
          QR 코드 히스토리 및 고급 기능을 이용하려면 로그인하세요
        </CardDescription>
      </CardHeader>
      <CardContent className="space-y-6">
        <GoogleSignInButton
          callbackUrl={callbackUrl}
          onSignIn={handleGoogleSignIn}
        />

        <RememberMeCheckbox
          checked={rememberMe}
          onCheckedChange={setRememberMe}
        />

        <ContinueWithoutSignIn />
      </CardContent>
    </Card>
  );
};
