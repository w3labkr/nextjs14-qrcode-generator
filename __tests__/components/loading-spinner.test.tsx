import { render } from "@testing-library/react";
import { LoadingSpinner } from "@/components/loading-spinner";

describe("LoadingSpinner", () => {
  it("로딩 스피너가 정상적으로 렌더링되어야 한다", () => {
    const { container } = render(<LoadingSpinner />);
    
    // 로딩 스피너 컨테이너 확인
    const spinner = container.querySelector(".animate-spin");
    expect(spinner).toBeInTheDocument();
    expect(spinner).toHaveClass("animate-spin", "rounded-full", "h-8", "w-8", "border-b-2", "border-primary");
  });

  it("적절한 스타일링 클래스를 가져야 한다", () => {
    const { container } = render(<LoadingSpinner />);
    
    // 외부 컨테이너 확인
    const outerContainer = container.querySelector(".flex.min-h-screen");
    expect(outerContainer).toBeInTheDocument();
    expect(outerContainer).toHaveClass("flex", "min-h-screen", "items-center", "justify-center");
  });

  it("로딩 스피너의 구조가 올바르게 렌더링되어야 한다", () => {
    const { container } = render(<LoadingSpinner />);
    
    // 전체 구조 확인
    expect(container.firstChild).toHaveClass("flex", "min-h-screen", "items-center", "justify-center");
    
    // 스피너 div 확인
    const spinner = container.querySelector(".animate-spin");
    expect(spinner).toBeInTheDocument();
  });

  it("스냅샷 테스트", () => {
    const { container } = render(<LoadingSpinner />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
