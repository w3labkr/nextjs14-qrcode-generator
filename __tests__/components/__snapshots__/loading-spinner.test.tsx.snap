// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoadingSpinner 스냅샷 테스트 1`] = `
<div
  class="flex min-h-screen items-center justify-center"
>
  <div
    class="animate-spin rounded-full h-8 w-8 border-b-2 border-primary"
  >
     
  </div>
</div>
`;
